packages/backend-common/src/reading/GitlabUrlReader.ts

```diff
@@ -256,4 +256,69 @@ export class GitlabUrlReader implements UrlReader {
     const { host, token } = this.integration.config;
     return `gitlab{host=${host},authed=${Boolean(token)}}`;
   }
+
+  async getGitlabFetchUrl(target: string): Promise<string> {
```

Comment:
Could we make all of the new added methods private? That way we don't need to change the public contract (leaving the api-report file unchanged). This is a bit nitpicky, but since anything we export might turn out to be called by basically anyone, we'll be making it a future breaking change to ever iterate on this contract even in the slightest :)

To be able to call them in tests, you can take the easy route of going via `any`; `(gitlabProcessor as any).somePrivateMethod(...` for example. Yeah it loses type safety within the test of course but it works.